tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,300 @@
+
+import joptsimple.OptionException;
+import joptsimple.OptionSpec;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaShareConsumer;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandDefaultOptions;
+import org.apache.kafka.server.util.CommandLineUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static joptsimple.util.RegexMatcher.regex;
+import static org.apache.kafka.tools.ConsumerPerformance.printConsumerProgress;
+
+public class ShareConsumerPerformance {
+    private static final Logger LOG = LoggerFactory.getLogger(ShareConsumerPerformance.class);
+
+    public static void main(String[] args) {
+        try {
+            LOG.info("Starting share consumer...");
+            ShareConsumerPerfOptions options = new ShareConsumerPerfOptions(args);
+            AtomicLong totalMessagesRead = new AtomicLong(0);
+            AtomicLong totalBytesRead = new AtomicLong(0);
+            AtomicLong joinTimeMs = new AtomicLong(0);
+            AtomicLong joinTimeMsInSingleRound = new AtomicLong(0);
+
+            if (!options.hideHeader())
+                printHeader(options.showDetailedStats());
+
+            KafkaShareConsumer<byte[], byte[]> shareConsumer = new KafkaShareConsumer<>(options.props());
+            long bytesRead = 0L;
+            long messagesRead = 0L;
+            long lastBytesRead = 0L;
+            long lastMessagesRead = 0L;
+            long currentTimeMs = System.currentTimeMillis();
+            long startMs = currentTimeMs;
+            consume(shareConsumer, options, totalMessagesRead, totalBytesRead, bytesRead, messagesRead, lastBytesRead,
+                    lastMessagesRead, joinTimeMsInSingleRound);
+            long endMs = System.currentTimeMillis();
+
+            Map<MetricName, ? extends Metric> metrics = null;
+            if (options.printMetrics())
+                metrics = shareConsumer.metrics();
+            shareConsumer.close();
+
+            // print final stats
+            double elapsedSec = (endMs - startMs) / 1_000.0;
+            long fetchTimeInMs = (endMs - startMs) - joinTimeMs.get();
+            if (!options.showDetailedStats()) {
+                double totalMbRead = (totalBytesRead.get() * 1.0) / (1024 * 1024);
+                System.out.printf("%s, %s, %.4f, %.4f, %d, %.4f, %d, %.4f, %.4f%n",
```

Comment:
Why don't we want to use the logger here?

Reply:
I think it's based on `ConsumerPerformance.java` and is just following that.

Reply:
I see, but may be it could be just logger here. We can control the output style.

Reply:
If we do that, we do it as a separate PR and we change `ConsumerPerformance` too.